tfac/figures/figure5.py

```diff
@@ -0,0 +1,171 @@
+from matplotlib.patches import Patch
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+from tfac.figures.figureCommon import getSetup
+from tfac.dataImport import form_tensor, import_cytokines, import_rna
+from tfac.predict import predict_known, predict_unknown
+from tfac.tensor import perform_CMTF
+
+OPTIMAL_SCALING = 32
+
+
+def makeFigure():
+    data_types, patient_data = get_data_types()
+    train_samples = run_cv(data_types, patient_data)
+    validation_samples = run_unknown(data_types, patient_data)
+    fig = plot_results(train_samples, validation_samples)
+
+    return fig
+
+
+def get_data_types():
+    """
+    Creates data for classification.
+
+    Parameters:
+        None
+
+    Returns:
+        data_types (list[tuple]): data sources and their names
+        patient_data (pandas.DataFrame): patient metadata
+    """
+    plasma_cyto, serum_cyto = import_cytokines()
+    tensor, matrix, patient_data = form_tensor(OPTIMAL_SCALING)
+    patient_data = patient_data.loc[:, ['status', 'type']]
+
+    components = perform_CMTF(tensor, matrix, 9)
```

Comment:
Leave off component number so that it uses the default, so this is defined in one place.